osidb/tests/test_flaw.py

```diff
@@ -941,3 +956,26 @@ def test_validate_cwe_format(self, cwe, should_raise):
             assert "CWE IDs is not well formated." in str(e)
         else:
             assert FlawFactory(cwe_id=cwe)
+
+    @pytest.mark.parametrize(
+        "embargoed_flaw,embargoed_tracker,should_raise",
+        [
+            (True, True, False),
+            (True, False, True),
+            (False, False, False),
+            (False, True, False),
+        ],
+    )
+    def test_validate_tracker_flaw_accesses(
+        self, embargoed_flaw, embargoed_tracker, should_raise
+    ):
+        """test Tracker model validator making sure flaws can't have a public tracker"""
+        affect = AffectFactory(flaw__embargoed=embargoed_flaw)
+        if should_raise:
+            with pytest.raises(
+                ValidationError,
+                match="Tracker is public but is associated with an embargoed flaw",
+            ):
+                TrackerFactory(affects=[affect], embargoed=embargoed_tracker)
+        else:
+            assert TrackerFactory(affects=[affect], embargoed=embargoed_tracker)
```

Comment:
We should have an extra test with > 1 flaw linked to a tracker, in which both are embargo, both are public and 1 embargo 1 public since the relationship is M2M

Reply:
added new test cases for M2M relationship.